src/plugins/ui_actions/public/actions/action_internal.ts

```diff
@@ -59,7 +59,7 @@ export class ActionInternal<A extends ActionDefinition = ActionDefinition>
     return this.definition.getIconType(context);
   }
 
-  public getDisplayName(context: Context<A>): string {
+  public getDisplayName(context: Context<A>): JSX.Element | string {
```

Comment:
Can we rename this since Display name is misleading here? `getDisplayElement` perhaps?

Reply:
Agreed that name would better, but I'd recommend against it at this point, because it would mean a lot of downstream changes as well (such as other plugins not in this repository).